contracts/strategies/traderjoe/TraderJoe.sol

```diff
@@ -245,41 +219,335 @@ contract TraderJoe is UUPSUpgradeable, StrategyOwnablePausableBaseUpgradeable {
         TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
             .getStorage();
 
-        assetValuations = new Valuation[](1);
-        assetValuations[0] = Valuation(
-            address(strategyStorage.lpToken),
-            (getTraderJoeLpBalance() * getTraderJoeLpReserve()) /
-                strategyStorage.lpToken.totalSupply()
-        );
+        uint256 binsAmount = strategyStorage.binIds.length;
+        assetValuations = new Valuation[](binsAmount);
+
+        for (uint256 i; i < binsAmount; ++i) {
+            uint256 lpTokenBalance = strategyStorage.lbPair.balanceOf(
+                address(this),
+                strategyStorage.binIds[i]
+            );
+
+            (uint256 reserveX, uint256 reserveY) = strategyStorage
+                .lbPair
+                .getBin(uint24(strategyStorage.binIds[i]));
+
+            uint256 totalSupply = strategyStorage.lbPair.totalSupply(
+                strategyStorage.binIds[i]
+            );
+
+            // Assume that USDC.e price equals to USDC price.
+            uint256 valuation = (lpTokenBalance * (reserveX + reserveY)) /
+                totalSupply;
+
+            assetValuations[i] = Valuation(
+                address(strategyStorage.lbPair),
+                valuation
+            );
+        }
     }
 
     function _getLiabilityValuations(bool, bool)
         internal
         view
         virtual
         override
-        returns (Valuation[] memory liabilityValuations)
+        returns (Valuation[] memory)
     {}
 
-    function getTraderJoeLpBalance() public view returns (uint256) {
+    function adjustBins(
+        uint256[] calldata binIds,
+        uint256[] calldata binAllocations
+    ) public onlyOwner {
+        __checkBinIdsAndAllocations(binIds, binAllocations);
+
+        TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
+            .getStorage();
+
+        // Withdraw from all bins.
+        uint256 depositTokenBefore = depositToken.balanceOf(address(this));
+
+        __withdraw(getInvestmentTokenSupply());
+
+        uint256 depositTokenAfter = depositToken.balanceOf(address(this));
+
+        // Set bin IDs and allocations to the given ones.
+        strategyStorage.binIds = binIds;
+        strategyStorage.binAllocations = binAllocations;
+
+        // Deposit into the new bins with the new allocations.
+        uint256 depositTokenIncrement = depositTokenAfter - depositTokenBefore;
+
+        __deposit(depositTokenIncrement, 0);
+    }
+
+    function __checkBinIdsAndAllocations(
+        uint256[] calldata binIds,
+        uint256[] calldata binAllocations
+    ) private pure {
+        uint256 binsAmount = binIds.length;
+        uint256 allocationsAmount = binAllocations.length;
```

Comment:
I think it should be callend Lengh and not Amount

Reply:
It matter of perspective. If you see each of them as an array of sth, then I think length is better. But if you consider them as the number of bins/allocations, then amount, count, etc are acceptable for me.